websites/H/HackTheBox/presence.ts

```diff
@@ -0,0 +1,183 @@
+const presences: Record<string, PresenceData> = {
+		"/challenges/retired": {
+			details: "Challenges",
+			state: "Browsing retired challenges",
+		},
+		"/challenges/todo": {
+			details: "Challenges",
+			state: "Browsing to-do list",
+		},
+		"/challenges": {
+			details: "Challenges",
+			state: "Browsing active challenges",
+		},
+		"/login": {
+			details: "Login",
+			state: "Logging in",
+		},
+		"/fortresses": {
+			details: "Fortresses",
+			state: "Browsing fortresses",
+		},
+		"/endgames": {
+			details: "Endgames",
+			state: "Browsing endgames",
+		},
+		"/starting-point": {
+			details: "Starting Point",
+			state: "Browsing starting points",
+		},
+		"/rankings": {
+			details: "Rankings",
+			state: "Browsing rankings",
+		},
+		"/register": {
+			details: "Register",
+			state: "Creating new account",
+		},
+		"/machines/list/unreleased": {
+			details: "Machines",
+			state: "Browsing scheduled machines",
+		},
+		"/machines/list/active": {
+			details: "Machines",
+			state: "Browsing active machines",
+		},
+		"/machines/list/retired": {
+			details: "Machines",
+			state: "Browsing retired machines",
+		},
+		"/machines/list/todo": {
+			details: "Machines",
+			state: "Browsing machines to-do list",
+		},
+		"/home": {
+			details: "Homepage",
+		},
+		"/machines/{}": {
+			details: "Playing machine",
+		},
+		"/challenges/{}": {
+			details: "Playing challenge",
+		},
+		"/profile/overview": {
+			details: "Profile",
+			state: "Browsing profile overview",
+		},
+		"/profile/settings": {
+			details: "Profile",
+			state: "Changing profile settings",
+		},
+		"/profile/subscriptions/plans": {
+			details: "Profile",
+			state: "Browsing subscriptions",
+		},
+		"/users/{}": {
+			details: "Looking at profile",
+		},
+		"/tracks": {
+			details: "Tracks",
+			state: "Browsing tracks",
+		},
+	},
+	presence = new Presence({
+		clientId: "1042105891681472595",
+	});
+
+function getHomePageDetails() {
+	return `${
+		document
+			.querySelectorAll(".text-left.font-size15.col")[0]
```

Comment:
`.querySelectorAll(...)[0]` is still being used a lot where it could be replaced by `.querySelector(...)`